core/src/main/java/org/infinispan/expiration/impl/ExpirationManagerImpl.java

```diff
@@ -198,13 +201,14 @@ private void handleInStoreExpiration(K key, V value, Metadata metadata) {
     * @param value
     * @param metadata
     */
-   private void deleteFromStoresAndNotify(K key, V value, Metadata metadata) {
+   private CompletionStage<Void> deleteFromStoresAndNotify(K key, V value, Metadata metadata) {
```

Comment:
Same thing, this is in preparation for async persistence layer, since I have to do some more interesting locking to support it. I guess for now I can make it not return a stage though.